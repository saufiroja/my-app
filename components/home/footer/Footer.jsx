import React from 'react';

import {
  Container,
  Grid,
  Typography,
  Box,
  TextField,
  Button,
} from '@mui/material';

const Footer = () => {
  return (
    <Box px={{ xs: 3, sm: 10 }} py={{ xs: 5, sm: 10 }}>
      <Container maxWidth='md'>
        <Grid container spacing={5}>
          <Grid item xs={12} sm={4}>
            <Typography variant='h5' className='font-body' pb={2}>
              Impostor
            </Typography>
            <Typography className='font-body'>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus
              veritatis debitis, reprehenderit doloremque aspernatur modi unde
              ut cupiditate quasi consequuntur.
            </Typography>
          </Grid>
          <Grid item xs={12} sm={4}>
            <Box borderBottom={1}>Contributor</Box>
            <Box>
              <Typography role="paragraf" className='font-body' pb={1} pt={1}>
                Bogiant
              </Typography>
            </Box>
            <Box>
              <Typography className='font-body' pb={1} pt={1}>
                Fikri
              </Typography>
            </Box>
            <Box>
              <Typography className='font-body' pb={1} pt={1}>
                Hasby
              </Typography>
            </Box>
            <Box>
              <Typography className='font-body' pb={1} pt={1}>
                Oja
              </Typography>
            </Box>
            <Box>
              <Typography className='font-body' pb={1} pt={1}>
                Ulfa
              </Typography>
            </Box>
          </Grid>
          <Grid item xs={12} sm={4}>
            <Box borderBottom={1}>Sletter</Box>
            <Typography className='font-body' pb={1} pt={1}>
              Enter Email Address
            </Typography>
            <Box pt={1}>
              <TextField
                id='outlined-basic'
                label='Email'
                variant='outlined'
                sx={{ width: '100%' }}
              />
            </Box>
            <Box pt={2}>
              <Button
                size='small'
                variant='outlined'
                className='font-body'
                sx={{ width: '100%' }}
              >
                Subscriber
              </Button>
            </Box>
          </Grid>
        </Grid>
        <Box
          className='font-body'
          textAlign='center'
          pt={{ xs: 5, sm: 10 }}
          pb={{ xs: 5, sm: 0 }}
        >
          &reg;{new Date().getFullYear()} Impostor Team. All Rights Reserved
        </Box>
      </Container>
    </Box>
  );
};

export default Footer;
