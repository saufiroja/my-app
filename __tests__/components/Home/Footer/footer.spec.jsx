import React from 'react'
import { render, screen } from '@testing-library/react'
import Footer from '../../../../components/home/footer/Footer'
import renderer from "react-test-renderer"

describe('Footer', () => {
    it('renders a heading', () => {
        render(<Footer />)

    const heading = screen.getByRole('heading', {
        name: "Impostor",
        })

        expect(heading).toBeInTheDocument()
    })

    it('matches the snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON()

    expect(tree).toMatchSnapshot()
    })
})