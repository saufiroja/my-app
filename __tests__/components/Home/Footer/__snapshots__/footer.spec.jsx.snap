// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches the snapshot 1`] = `
<div
  className="MuiBox-root css-td4wtc"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 font-body css-1ux20dk-MuiTypography-root"
        >
          Impostor
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1 font-body css-ahj2mt-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus veritatis debitis, reprehenderit doloremque aspernatur modi unde ut cupiditate quasi consequuntur.
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-vcv939"
        >
          Contributor
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 font-body css-1fl9go7-MuiTypography-root"
            role="paragraf"
          >
            Bogiant
          </p>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 font-body css-1fl9go7-MuiTypography-root"
          >
            Fikri
          </p>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 font-body css-1fl9go7-MuiTypography-root"
          >
            Hasby
          </p>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 font-body css-1fl9go7-MuiTypography-root"
          >
            Oja
          </p>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 font-body css-1fl9go7-MuiTypography-root"
          >
            Ulfa
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-vcv939"
        >
          Sletter
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1 font-body css-1fl9go7-MuiTypography-root"
        >
          Enter Email Address
        </p>
        <div
          className="MuiBox-root css-zoser8"
        >
          <div
            className="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="outlined-basic"
              id="outlined-basic-label"
            >
              Email
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="outlined-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-173wfxe"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root css-1v3caum"
        >
          <button
            className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root font-body css-1jb5gvc-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Subscriber
          </button>
        </div>
      </div>
    </div>
    <div
      className="font-body MuiBox-root css-1gjlakx"
    >
      Â®
      2022
       Impostor Team. All Rights Reserved
    </div>
  </div>
</div>
`;
